microsoft-edge/devtools-guide-chromium/crash-analyzer/index.md

```diff
@@ -0,0 +1,338 @@
+| The **Open Symbol Server settings** (![the 'Open Symbol Server settings' icon](./index-images/open-symbol-server-settings-icon.png)) button | Display or modify the Symbol Server settings. |
+| The **How to use** (![the 'How to use' icon](./index-images/how-to-use-icon.png)) button | Opens the present article. |
+
+
+<!-- ====================================================================== -->
+## Storing source maps using Azure Artifacts Symbol Server
+
+The **Crash analyzer** tool works best if your source maps are securely stored using Azure Artifacts Symbol Server.  This is because source maps are retrieved on demand when analyzing your error, and the exact version which produced the error message might not be cached on your development machine.
+
+See [Securely debug original code by publishing source maps to the Azure Artifacts symbol server](../javascript/publish-source-maps-to-azure.md).
+
+
+<!-- ====================================================================== -->
+## Required stack trace format
+
+Your code must produce a stack trace in the format conventionally produced by the V8 JavaScript engine, with additional metadata about the scripts being executed.  Here's an example stack trace that was produced by Microsoft Edge DevTools:
+
+```
+Error: Could not instantiate model for frameId
+    at Ei (devtools://devtools/bundled/panels/elements/elements.js:1:6669)
+    at vi (devtools://devtools/bundled/panels/elements/elements.js:1:6756)
+    at devtools://devtools/bundled/panels/elements/elements.js:1:7403
+    at async Object.children (devtools://devtools/bundled/panels/elements/elements.js:1:7137)
+    at async b (devtools://devtools/bundled/ui/components/tree_outline/tree_outline.js:1:4438)
+    at async #v (devtools://devtools/bundled/ui/components/tree_outline/tree_outline.js:1:8182)
+
+Source modules:
+    devtools://devtools/bundled/panels/elements/elements.js 22994ef71c62ba25b9494e26da78b621c5995bff1830c183035b4c1eb7c6b12b
+    devtools://devtools/bundled/ui/components/tree_outline/tree_outline.js bac7011989ae5c7870c7f2964e88a51306eb2d520236329f09be5f5850105a42
+```
+
+The additional metadata is the line `Source modules:` and the lines below it.  Each line in that section is parsed as follows:
+
+* The line must end with a 64-character line of hexadecimal characters corresponding to the SHA-256 hash of the script being executed.  This is a byte-for-byte hash of the content for each JavaScript file.
+
+* The rest of the line is trimmed and treated as the module name, such as `/panels/elements/elements.js`.  The module name should correspond to file names listed in the main part of the stack trace.
+
+
+<!-- ====================================================================== -->
+## Producing the required stack trace format
```

Comment:
This section feels unnecessary. The NPM library from the **Reporting non-fatal JavaScript exceptions or similar data to tools** section already contains this code. The expected steps for users is (as far as I can understand): 

1. Use the NPM package in your web app. The package produces stack traces formatted as the tool expects them to be. Perhaps we need a section that explains how to use the library in this NPM package. Specifically, web developers should import the library and call `installErrorStackModuleAnnotations(Error);`.
2. Put in place some kind of production stack trace collection system, such as Application Insights. This way, when your web app crashes on a customer's machine, the stack trace that got thrown when the crash occurred is recorded in your system somewhere.
3. Then, later, use the Crash Analyzer tool to "unminify" the stack traces you collected in production, which are specially formatted. This step makes use of the Symbol Server configuration to do the unminification of the stack trace. Thanks to this, web devs can see exactly where the crash happened in the code, as if they were on the customer's machine right when the crash occurred.

Reply:
Proposed structure for the article:

1. Intro (what is the tool, how we expect it to be used)
2. How to collect crash stack traces in production (explain how to use the NPM package and link to Application Insights).
3. Analyze crash stack traces in the tool
   1. Open the tool
   2. UI items
   3. Use the tool
4. Provide feedback
5. See also

Reply:
Resulting outline from restructuring article guided by the above outline: 
```
# Crash analyzer tool
## Storing source maps using Azure Artifacts Symbol Server
## How to use the npm package
## Link to Application Insights
## Open the Crash analyzer tool
#### From the More tools menu
#### From the Command Menu
## UI features
## Analyze crash stack traces in the Crash analyzer tool
## Reporting non-fatal JavaScript exceptions or similar data to tools
## Required stack trace format
## Provide feedback
## See also
```